components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesHistoryStorage.kt

```diff
@@ -247,6 +253,29 @@ open class PlacesHistoryStorage(
         }
     }
 
+    /**
+     * Enqueues periodic storage maintenance worker to WorkManager that prunes database entries
+     * when it exceeds [PlacesStorageMaintenanceWorker.DB_SIZE_LIMIT_IN_BYTES].
+     */
+    override fun registerStorageMaintenanceWorker() {
+        WorkManager.getInstance(context).enqueueUniquePeriodicWork(
+            PlacesStorageMaintenanceWorker.UNIQUE_NAME,
+            ExistingPeriodicWorkPolicy.KEEP,
+            periodicStorageWorkRequest<PlacesStorageMaintenanceWorker>(
+                repeatInterval = PlacesStorageMaintenanceWorker.WORKER_PERIOD_IN_HOURS,
+                repeatIntervalTimeUnit = TimeUnit.HOURS,
+                tag = PlacesStorageMaintenanceWorker.UNIQUE_NAME,
+            ) {
```

Comment:
nit: I think you can remove this now since you now have reasonable defaults on this extension:
```suggestion
            periodicStorageWorkRequest<PlacesStorageMaintenanceWorker>(
                tag = PlacesStorageMaintenanceWorker.UNIQUE_NAME,
            ) {
```